### PR TITLE
templatize timedelta arith ops

### PR DESCRIPTION
**Pull request by jbrockmendel**
Some Timedelta operations are constructed using `_binary_op_method_timedeltalike`, while a bunch of others re-implement shared logic.  This PR is an attempt to get rid of the small differences in this shared logic so they can eventually use a single template.

### CODE REVIEW

File: pandas/_libs/tslibs/timedeltas.pyx [hunk 1]
```diff
@@ -471,9 +471,12 @@ def _binary_op_method_timedeltalike(op, name):
     # define a binary operation that only works if the other argument is
     # timedelta like or an array of timedeltalike
     def f(self, other):
-        if hasattr(other, 'delta') and not PyDelta_Check(other):
-            # offsets.Tick
-            return op(self, other.delta)
+        if hasattr(other, '_typ'):
```
Comment (jreback) [inline pandas/_libs/tslibs/timedeltas.pyx]: rather than repeating this code can you make a function?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/tslibs/timedeltas.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/timedeltas.pyx]: thats exactly what this PR is preliminary to, but with more than just these 3 lines.

Comment (jreback) [reply pandas/_libs/tslibs/timedeltas.pyx]: oh ok

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**